cat/src/main.rs

```diff
@@ -0,0 +1,154 @@
+use std::{
+    fs::File,
+    io::{prelude::*, stdin, BufReader, ErrorKind},
+};
+
+use clap::{load_yaml, App, AppSettings::ColoredHelp, ArgMatches};
+
+fn main() -> std::io::Result<()> {
```

Comment:
Main return Result it's not the right call here (well, at least till Teminator trait hits stable), when `Err`, code will exit with the wrong exit code and with not the format coreutils tools should display